application/src/main/java/org/togetherjava/tjbot/commands/moderation/UnmuteCommand.java

```diff
@@ -0,0 +1,130 @@
+ * also try to DM the user to inform them about the action and the reason.
+ * <p>
+ * The command fails if the user triggering it is lacking permissions to either unmute other users
+ * or to unmute the specific given user (for example a moderator attempting to unmute an admin).
+ */
+public final class UnmuteCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(UnmuteCommand.class);
+    private static final String TARGET_OPTION = "user";
+    private static final String REASON_OPTION = "reason";
+    private static final String COMMAND_NAME = "unmute";
+    private static final String ACTION_VERB = "unmute";
+    private final Predicate<String> hasRequiredRole;
+    private final Predicate<String> isMuteRole;
+
+    /**
+     * Constructs an instance.
+     */
+    public UnmuteCommand() {
+        super(COMMAND_NAME,
+                "Unmutes the given already muted user so that they can send messages again",
+                SlashCommandVisibility.GUILD);
+
+        getData().addOption(OptionType.USER, TARGET_OPTION, "The user who you want to unmute", true)
+            .addOption(OptionType.STRING, REASON_OPTION, "Why the user should be unmuted", true);
+
+        hasRequiredRole = Pattern.compile(Config.getInstance().getSoftModerationRolePattern())
+            .asMatchPredicate();
+        isMuteRole = Pattern.compile(Config.getInstance().getMutedRolePattern()).asMatchPredicate();
+    }
+
+    private static void handleNotMutedTarget(@NotNull Interaction event) {
+        event.reply("The user is not muted.").setEphemeral(true).queue();
+    }
+
+    private void unmuteUser(@NotNull Member target, @NotNull Member author, @NotNull String reason,
+            @NotNull Guild guild, @NotNull SlashCommandEvent event) {
+        User targetUser = target.getUser();
+        String dmMessage = """
+                Hey there, you have been unmuted in the server %s.
+                This means you can now send messages in the server again.
+                The reason for the unmute is: %s
+                """.formatted(guild.getName(), reason);
+
+        event.getJDA()
+            .openPrivateChannelById(targetUser.getId())
+            .flatMap(channel -> channel.sendMessage(dmMessage))
+            .mapToResult()
+            .flatMap(sendDmResult -> {
+                logger.info("'{}' ({}) unmuted the user '{}' ({}) in guild '{}' for reason '{}'.",
+                        author.getUser().getAsTag(), author.getId(), targetUser.getAsTag(),
+                        targetUser.getId(), guild.getName(), reason);
+
+                return guild.removeRoleFromMember(target, getMutedRole(guild).orElseThrow())
+                    .reason(reason)
+                    .map(unmuteResult -> sendDmResult.isSuccess());
+            })
+            .map(hasSentDm -> {
+                String dmNotice =
+                        Boolean.TRUE.equals(hasSentDm) ? "" : "(Unable to send them a DM.)";
```

Comment:
[See my comment on the kick, ban and unban command](https://github.com/Together-Java/TJ-Bot/pull/244#discussion_r748922925)

Reply:
sure, will do

Reply:
done.